src/snowflake/cli/plugins/nativeapp/run_processor.py

```diff
@@ -146,6 +146,7 @@ def get_existing_version_info(self, version: str) -> Optional[dict]:
                     version,
                     name_col=VERSION_COL,
                     in_clause=f"in application package {self.package_name}",
+                    pick_if_multiple=lambda rows: rows[-1],
```

Comment:
I forget the output rules here, why are we only returning the last row? Is some order guaranteed in the output that `-1` will always be deterministic?

Reply:
I had made an assumption here that the patches were in order, and we'd be interested in the last patch. Though now I'm thinking about it I don't think there's any such guarantees stated. Right now, everything that calls `get_existing_version_info` just throws away the result after it determines if it's `None` or not, so it's moot -- which is why I'm kind of thinking maybe we don't need such a deep change and can just call `show versions like <version name>` directly...

WDYT?

Reply:
I like the idea here to keep it simple, I think `show_specific_object` is probably an overkill for our kind of query here. We could implement our own python `max` on the patch column for a given version to really determine the max patch seen so far.

[if only this could be treated as regular sql :D ]
```
with t1 as (show versions like <> in application package)
select max(patch) from t1
```

Reply:
OK, if you feel the same way then it's settled, I'll go for a simpler implementation and ping you again when this is ready for re-review.